stub/src/main/java/io/grpc/kotlin/ClientCalls.kt

```diff
@@ -285,7 +285,10 @@ object ClientCalls {
       clientCall.start(
         object : ClientCall.Listener<ResponseT>() {
           override fun onMessage(message: ResponseT) {
-            if (!responses.offer(message)) {
+            try {
+              responses.trySend(message).getOrThrow()
+            }
+            catch (e: Exception) {
               throw AssertionError("onMessage should never be called until responses is ready")
             }
```

Comment:
Similar to the other comment I think the code that would match previous semantics would be

```kotlin
            responses.trySend(message).onFailure { ex ->
              throw ex ?: AssertionError("onMessage should never be called until responses is ready")
            }
```